src/app/operations/intraledger_payment.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Result;
+
+use rust_decimal::Decimal;
+
+use crate::{app::App, client::types::Wallet, errors::payment_error::PaymentError};
+
+impl App {
+    pub async fn intraledger_payment(
+        &self,
+        username: String,
+        wallet: Wallet,
+        cents: Option<Decimal>,
+        sats: Option<Decimal>,
+        memo: Option<String>,
+    ) -> Result<()> {
+        if wallet == Wallet::Btc && sats.is_none() {
+            return Err(PaymentError::AmountNotSpecifiedBTC.into());
```

Comment:
Non optional arguments should be validated in the cli. Clap has good primitives for this that cover most needs.

Reply:
Thank you for this, @bodymindarts. I fixed this in commit 7e229af6ad889eb9bc0791aa572444ce6d9ac221. It significantly streamlined the code.